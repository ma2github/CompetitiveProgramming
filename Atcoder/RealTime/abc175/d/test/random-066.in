1168 3022594
641 408 230 730 103 831 895 247 267 513 30 1020 636 226 771 224 242 176 1113 1054 62 511 672 944 94 622 903 314 161 837 742 1052 1142 689 318 810 1090 12 866 55 255 158 491 166 286 1036 818 876 353 779 403 913 1061 97 265 293 69 1089 9 709 1064 279 500 326 36 871 110 1056 116 1051 897 905 86 615 549 241 646 67 144 1164 343 629 705 475 822 348 437 276 306 638 316 757 350 195 386 26 723 620 700 593 772 191 90 13 859 366 1063 257 291 38 658 777 355 322 719 457 1105 1148 718 169 240 1092 309 627 616 907 341 856 498 1117 299 707 308 20 201 894 799 461 107 889 274 945 151 1039 936 1091 447 1027 764 552 783 715 468 377 134 541 141 73 78 1126 212 489 1082 16 659 531 635 565 1005 374 904 180 42 66 325 303 644 207 724 1081 388 452 969 601 760 1006 453 993 1022 600 745 402 65 256 109 645 1014 688 628 260 521 148 1128 1158 1070 791 1116 250 979 890 307 1028 1153 420 220 1119 357 667 920 788 474 57 650 746 867 221 959 881 507 6 538 490 1037 781 664 962 1003 427 533 663 1025 562 1019 770 673 172 152 138 502 487 656 23 45 862 1002 626 1058 933 296 661 81 119 383 131 1133 981 986 749 80 145 997 864 512 896 111 517 938 310 203 89 765 300 918 378 984 1007 642 231 344 123 815 536 422 643 775 611 298 1093 954 14 442 462 470 235 898 254 848 1156 682 1107 1024 666 971 543 972 127 941 283 819 133 11 758 416 1021 967 750 559 496 312 566 949 906 768 950 965 184 568 473 177 527 703 916 922 368 209 523 476 694 47 159 934 469 987 713 302 710 170 262 1101 684 358 339 1047 313 121 801 1098 1013 448 612 398 1125 1017 914 486 64 569 101 10 91 1016 966 112 697 17 853 1130 613 1030 140 331 126 395 1067 414 219 467 1060 346 465 792 429 880 633 1031 787 624 335 503 1059 598 911 506 304 70 576 1127 968 1114 1094 1095 542 199 40 1108 392 264 575 411 1041 1121 1112 135 129 217 482 956 1131 782 963 1161 347 729 983 435 494 485 930 1073 586 58 125 1075 995 529 114 106 352 725 1124 769 362 942 410 826 381 218 820 665 1076 252 1100 932 405 200 1029 599 556 621 834 816 877 796 637 609 957 185 654 179 657 546 438 1080 953 75 415 878 990 320 1157 1151 693 329 409 785 413 505 863 49 737 901 399 1085 137 1147 698 630 721 888 558 929 424 371 537 1012 883 282 858 1045 367 369 935 631 295 76 193 269 923 861 213 550 847 989 597 432 669 84 824 1132 458 855 380 1066 988 25 1044 731 1168 651 88 124 454 773 1115 439 365 270 108 544 836 728 887 198 685 1134 426 143 43 197 879 82 1033 418 766 1167 991 471 271 1038 662 960 515 564 579 525 632 263 1046 740 735 755 1072 285 1111 596 917 417 393 1032 572 675 164 1009 996 833 851 477 168 104 174 1048 436 50 281 514 443 961 31 206 157 79 51 683 85 428 292 1034 32 244 805 376 1088 5 175 259 460 561 92 902 1123 361 1140 53 253 334 854 472 382 33 210 763 670 204 1026 434 332 534 528 909 940 59 154 456 1120 122 1008 323 1053 553 483 747 1096 1078 868 142 1071 1055 674 739 60 87 354 952 992 1050 817 345 223 634 389 28 115 24 29 233 652 338 639 182 678 497 608 95 618 681 899 297 1110 860 444 238 21 886 921 294 15 1150 117 196 1040 1062 741 671 1129 128 680 910 384 555 44 640 522 349 243 827 130 370 234 518 54 823 852 610 560 508 582 328 391 96 1136 407 947 1155 841 939 784 982 1139 171 619 397 845 774 1122 100 563 647 150 951 46 870 248 809 188 337 56 480 390 1083 699 882 1001 727 363 893 139 744 1079 1099 832 754 835 668 577 356 499 676 364 846 385 795 34 532 1087 603 431 311 797 975 708 387 412 720 317 153 595 202 333 189 756 545 573 190 530 1141 278 686 1146 72 733 120 738 1154 687 706 926 617 261 994 449 830 258 3 340 419 928 839 571 1084 548 1065 752 1015 479 105 501 1104 22 648 225 155 99 874 516 214 811 591 605 807 973 289 524 37 194 1004 588 653 1102 780 726 716 330 1138 798 574 690 1103 268 249 272 1109 208 1069 375 1011 840 41 481 466 759 998 736 584 478 216 570 900 679 287 1010 147 751 535 132 239 273 594 183 68 1074 655 464 691 908 404 825 704 488 649 604 712 857 379 1042 315 342 974 441 280 1023 228 324 714 77 526 743 1077 869 821 885 814 173 35 1137 423 48 925 98 93 812 592 215 872 61 136 955 321 290 351 842 850 162 1057 1163 1018 1118 625 794 717 277 583 113 838 495 802 8 692 245 451 1145 425 492 421 843 547 102 118 551 52 931 160 360 7 924 288 761 587 509 1135 808 205 937 804 275 711 828 813 1149 614 520 1097 1160 580 165 237 1 849 589 327 891 540 884 803 964 919 227 539 401 440 229 722 455 829 790 1165 1049 246 406 1086 305 875 187 1043 211 178 999 83 1068 606 581 865 943 493 430 892 946 181 156 149 222 446 27 186 554 372 167 602 607 1166 39 359 732 762 433 767 63 977 510 394 696 463 1000 18 445 702 400 484 2 504 163 585 4 74 701 232 912 336 284 778 19 844 1144 266 251 578 786 71 1106 146 958 985 623 915 519 557 948 748 800 677 806 980 373 976 660 590 776 1152 236 567 1035 695 459 970 450 192 978 1143 873 793 301 753 789 1159 396 319 734 1162 927
717634236 -703822874 -511820537 958911360 -568327552 876522882 -403478663 -252496599 -142297163 902882104 841521460 471859195 -849954333 -807652591 -808570652 725738598 -740107574 461564847 -821949606 -399044563 -850201676 -81692569 -710095799 -549424754 -202958397 924090672 519292237 -589031870 -587317405 804062659 -13779546 158385971 986154179 -660737259 35189948 -721311368 -341257230 -371727278 -881045612 492476604 -365533268 303971807 -219402408 534690036 736546854 837821271 -590742383 -258335362 -690385124 588388243 -683140744 563809412 155328756 234360049 753886794 -467109401 286037573 132442642 -424397226 -950500627 -386181986 -860093669 -857820246 -165139229 -499632636 -151851540 -522001079 235130808 541210063 -745052260 -686347132 580650400 845558796 46475734 -480645405 -299149355 920760387 -420130542 7966001 449839758 -989530611 711596381 -505790343 -892850141 945523862 -837472085 987399736 662942093 -421859882 377287712 -910306442 -837987609 323360534 204182824 933348544 275430838 -878324727 -355981900 507423478 929766554 -128981228 911492375 -802932747 -631191726 -390663257 -215667515 178119045 -846284477 -979165162 976008867 -221854645 512810481 54146935 819153236 -876775527 -494026344 -93035180 -724179164 -503408983 276214799 -467676800 170311577 997366199 199059502 -138491594 632435876 -151417633 969534672 511436880 -819654144 -133796207 950259074 -328587612 268734257 851319840 450101271 -369904535 839720213 630737500 -589072865 -521130910 -335316006 574122800 -397457291 -243282879 335143738 191752440 769314813 971624671 -824203940 764863381 -468880739 549945749 18343588 433903408 -908733335 -685174392 -62764942 714746718 459592225 573023418 -543349929 -920347375 -632122292 180808782 -482206227 -571051543 489648998 -342961901 586459370 -690107018 -911128084 409356982 -577006149 578593416 578391825 -780251467 361930797 821968128 490353585 -582013667 -979527178 621210310 177775852 -878802608 495002291 -375106500 768565743 923321461 69855169 624739450 -225107817 374298940 -468593350 484637416 13176874 -220563153 76202508 -589529275 443591712 -881192579 380659890 728459073 -698518899 -425422624 -665128611 112921290 -665242593 50139598 670023988 56174332 -48863565 622765087 -76540571 50985824 -554355056 502729435 -906607106 -76334009 263041304 -154309478 651832876 537819225 -913547367 -703760442 360978496 626567473 -601023340 995530688 436766109 -63829370 53867780 791804201 912534261 861161254 709677861 -936430238 -922098428 940708280 -612037688 984168263 -242382912 -723509200 -29575509 920786238 -5334798 564395256 865529678 201784741 -189106045 -130581563 79487689 -424424279 -298642726 -49024071 157762984 260852976 308717608 -628782443 836100487 -420351007 988356937 -953690510 -59132563 793091118 77452180 368755994 -410364694 839274172 608559318 -285864600 763007870 23382526 -884118163 481104974 -467841623 -90346010 -269467164 500463558 -384811463 -472011867 -503236491 817795461 -195856843 7463486 750093693 -301984369 311211305 441032606 -522521763 -584727078 44319132 41927708 -248477344 -632521980 352761468 177713825 815401846 860333003 780332296 -602753081 6022372 -215344580 -20695445 742021593 435276552 323787943 160394189 296012446 -152318682 -420389570 478186861 524605943 376701842 656043573 -628348868 -417810742 -896769264 947085384 -56558924 -110441493 958228272 -166397355 907809850 -708081665 -844456086 167869883 -931541285 524830544 561421624 -43475907 391631591 -598071488 -297401807 627581937 -728838388 -209531049 965055189 745365092 92879069 334660335 855446400 816219860 928474905 -214043076 825827730 -706623347 232454529 442033106 -878987519 -569531064 -736203562 -175547637 -764334601 817518378 -792226835 717247573 579151545 977910594 45987028 -299208865 294155209 -120878092 -360511670 -624358773 -806164734 479606426 90869269 -347828591 -201947956 -584316051 532864043 793497004 -268751376 870502562 -273155200 77849955 25426100 -546022631 868919526 -256115865 976609004 -686800679 250507702 -790040880 -45750937 -651167349 -986223294 -390759789 745847053 -690235052 525162570 -284175673 355888623 -822208918 -69671851 -219840595 114704667 -915939666 -464676735 893686004 808613805 56269453 849570528 -180606201 -940018935 727271562 585625429 264176956 -628246337 104511378 -25102519 10636767 -860626479 381246594 -502726302 -459669208 -21338707 -597204356 -140488174 -118198058 101650060 752054310 939381234 482331166 -747402309 947787841 -604017584 187659123 -204345658 -998550466 -837921627 195622303 -909813338 -426784303 506207880 -334235934 -939011715 -276902384 671660586 270018548 579795351 -879439913 163339537 -191067819 901156982 -664630077 945584884 541302317 424801102 211031259 251172065 618677222 305154820 146655941 -376192333 75611196 -275436593 -210088628 106781396 -909081489 -489710845 238802980 353184188 -712664890 -168983420 -116274302 -350073656 750210860 612461452 257852370 902064749 296031627 -385490275 -371976973 887602577 -232522568 452308212 -443367100 808061153 -216527033 601968118 -350333730 296669458 677434073 915175162 -628401531 -870916907 540041023 -8329209 -282800855 -12997734 745876831 -666876367 -952637195 92946834 -85905503 -683898081 -962689489 106498832 500621380 382220239 -730749094 -822227839 312897998 -152554757 218070514 -512675165 -290857528 158850012 629381183 226682701 422639401 -780826762 107063032 -457948338 -513596994 -913651753 47378137 385726347 45880976 -315284553 151740392 813541926 -575741666 -790605109 729372311 154349887 -544061540 658062993 555793974 60229574 -549979360 17229148 945069096 -664885857 398705338 -643098123 -17499115 400997222 784467878 -641725379 -271784409 -322296000 763993082 -820873809 623597645 227375088 -310786870 629305493 583897064 -602357706 -245021290 -510411260 530429060 -70585741 484777477 934681925 -544082369 339153141 -440730455 681273986 699442000 -721373268 -796869505 880178365 -67166341 -370897198 521077316 58159713 822226448 -845191845 309399933 -285864427 886665082 -930311198 550338373 -554536685 882641704 -285687570 -566769705 748048788 -994512403 388840179 992098562 -361891823 862110432 -666780835 620532186 -558986406 713128532 -336885954 -939244429 622256391 916241729 -20731389 22461644 561575696 676045317 -140869087 1770514 879160784 356333810 208015023 932632201 745434209 571092714 212462388 -881604573 -739056890 894473893 860085765 -133867145 -134318050 -397905161 -693109812 -575861459 896060215 983886241 743283330 306818829 -484808026 601280082 302294155 -310205020 -839485389 833259073 -652233656 712518802 -132416665 -865295478 217276943 -305978812 843580736 -16789275 -179613601 898630257 -550216795 174784837 386271267 835607887 -766411548 -119811315 -481643842 -289937000 118699087 537070870 -520203675 -219334409 -284449111 556894408 -199555426 918328867 32171560 8347976 633362817 -364352961 200575159 -509679324 240325426 711438121 -649965989 -827123068 581087754 894056257 137549897 692410721 -363968502 -615200210 -954433154 32861648 485319900 495913419 569751963 545468266 215014210 107780538 816757427 242932669 -754885290 -597364355 -864834528 -476402578 -719602467 109529628 -49112231 -728733613 843586071 -117887876 -25276458 -78998038 -618492954 -978511463 121841406 526170409 849835725 750248592 43360132 314798303 367814189 -799209702 440447266 148963975 -701643395 -814390084 47895451 787004656 842100882 -77124500 182543655 894678490 818819657 -984672722 -898956479 -577366364 -452067876 22020526 745802960 -459527478 -559880403 437760580 -427559937 -166258010 204446134 -247104432 844939929 -476831246 547285260 -669743221 66815953 -804301399 565014237 764183958 347981232 -787536050 701356666 378120551 275013624 -437286302 185059378 497563893 -115471392 -50039239 -18258606 -496446696 374881873 813260475 363991507 564224503 -785633499 955524224 90378945 822076688 972474175 -915193473 -753070743 -625667128 457593678 -803093118 646399462 -742665880 201671720 130420388 933798302 -590738960 583033003 573810847 478124545 217737289 296210357 -729674767 240857136 937374958 -643720227 673963053 -950398811 -899970335 450457372 -827648164 992631255 134874219 -387275379 133080839 -715218541 819311825 -645153894 851988938 -636467315 533936152 -546408874 -554514525 -59189486 -828366358 -44057367 265872975 101319811 -16210078 356970516 -805112321 -136414493 239836691 92652983 166189165 -713858962 -806791515 -414912784 259369721 352936224 735111597 -990819644 438522722 -920969494 -427255429 908263165 289471540 318464963 -925308987 668757739 117150854 849065620 -380151399 -54541549 612013646 -610396091 350499462 -778096835 81398335 771251348 -357213245 249387231 115183477 -902268268 950909948 -308827392 738818725 -217484062 860001367 -843270297 -286587195 -36908581 198677766 -717969964 -550072247 935910990 -167064665 810418084 -233926641 61365205 -986772348 -51570775 -91633265 -783852086 -693336991 8154330 -538993727 255659503 204958681 461872307 599372505 -359171959 759627101 -544425800 -284671924 -554125222 -12735174 -363999540 980408574 -971513323 240365732 -910288538 864258340 -5635829 126087399 -661893245 -987514191 -361706369 745821310 -932462338 -829661250 359768343 704657623 -257190426 547765184 -824582765 886271478 -537089722 -913326636 945758758 -382661683 -660854215 -222858487 337799984 332922742 966017003 -980460783 485103268 -155728332 -386562306 15995238 385411958 -486699209 -321362022 -298982462 639143590 -951499687 180948644 -237119458 -938227389 -159196808 -6687068 -370447639 850132254 -631236275 273902300 71797091 -946790800 780378045 -12030098 420676468 -876883533 -335587394 -425100244 -647059779 -198071821 -946076854 -712392480 -257412883 798476239 -561697491 -883698063 314591072 376756787 870207836 -694342753 -798169835 -37952837 -571834077 -915318164 -737715741 723953669 77185185 846461942 -833023608 601242137 -923588636 698647861 958899749 -627653260 334898326 296303115 -140725000 532056496 853206667 133462415 865571490 429256412 756908795 879419054 -353818755 -319933816 -6331829 388733893 811103638 -694136244 285906451 -407826640 -840773321 832678720 -118193487 -848359803 -67479400 -775388309 686298763 -572713154 455420185 -885203997 6521347 37810357 830092972 -276379866 -210966050 219631433 450167205 -490952001 516143954 -569228193 895123500 -200078600 -435630931 117327911 -484808048 670487283 851694766 336055798 -342098510 -196951737 -558289085 -52282688 301948606 426245075 -673164134 -806711916 -434924448 888618018 103638953 -944685102 -11052836 -344016198 944162097 186205008 -625296319 874802902 676982447 813973809 -436523048 164860941 745151699 -440191398 -419505 -412852475 156620546 -216612410 175383695 768173399 266310127 -717565810 37966638 998347392 -117447576 729381714 823138098 628807141 443626813 -15481982 599553976 21954901 830665828 685825421 -976891591 817538463 275168667 -918438636 -677423389 740241106 -459756579 -196497896 883375138 66907015 -275289126 689618712 852315414 -536304446 -343518886 -825559820 -528587080 -645370355 839074841 -47640515 961530138 -832768907 124048933 648729098 -551413217 -610048091 274056819 -750636620 873353915 -772060992 297757002 424090296 602647509 949704344 424878678 511270629 -47091558 908045306 50390350 -209891556 443474783 651514880 322237008 4447418 817240685 -919263149 -953131755 144724427 -501755708 -137310569 -979265242 -977917585 -957038280 -430718389 -578421653 -218893790 497779591 54172762 346638056 900139419 -884193329 -938416931 -114026288 -581445107 -424838942 -293158962 -782452365 -260769367 701982094 -256965202 -711112215 -140725829 157980256 682436015 -859351457 -809357007 -181639486 -470293322 609807708 578184570 -562844252 -810926105 -109618970 398955803 982891897 -71854407 451183994 -612412225 257852080 -603228957 -986779902 -535374322 915157103 -529518612 605729523 605913876 757585082 406347844 69602286 607630231 -270773252 -974848530 -818142172 -923625004 456741037 568866384 -568796609 590897887 128786340 479947067 99389672 997059026 53225021 -342133980 367327648 -553239259 399277662 904513350 -301327407 -322313384 -294781653 930770806 -458517738 -96189616 1778043 91813670 -842390899 -786007378 792365700 -845314988
