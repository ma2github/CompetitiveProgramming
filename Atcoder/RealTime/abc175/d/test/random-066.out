1974294332386279
